sound/soc/sof/intel/hda-dai-ops.c

```diff
@@ -140,9 +140,27 @@ static void hda_stream_put_dma(struct snd_sof_dev *sdev, struct snd_soc_dai *cpu
 	snd_hdac_ext_stream_release(hext_stream, HDAC_EXT_STREAM_TYPE_LINK);
 }
 
+static int hda_dma_setup(struct snd_sof_dev *sdev, void *dma_data, unsigned int format_val)
+{
+	struct hdac_ext_stream *hext_stream = dma_data;
+
+	snd_hdac_ext_stream_setup(hext_stream, format_val);
+
+	return 0;
+}
+
+static void hda_dma_free(struct snd_sof_dev *sdev, void *dma_data)
+{
+	struct hdac_ext_stream *hext_stream = dma_data;
+
+	snd_hdac_ext_stream_reset(hext_stream);
+}
+
```

Comment:
again I think we want to keep reset and setup in this order. Let's not try to think to hard on elegance and symmetry, this isn't how the code is used today.